src/App.tsx

```diff
@@ -1,27 +1,75 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  function usePromiseState(
```

Comment:
That's very nice you know how to create custom hooks, impressive :+1: (but it's better to do it outside a component)
Anyway, it's overcoding here. Also, you are doing 3 request fetch on the first render, but you should do it on the button click.